public/tabs/history/chat_history_page.tsx

```diff
@@ -96,14 +100,29 @@ export const ChatHistoryPage: React.FC<ChatHistoryPageProps> = React.memo((props
     [searchName]
   );
 
-  useEffect(() => {
-    if (props.shouldRefresh) services.conversations.reload();
+  useUpdateEffect(() => {
+    if (!props.shouldRefresh) {
+      return;
+    }
+    services.conversations.reload();
+    return () => {
+      services.conversations.abortController?.abort();
+    };
   }, [props.shouldRefresh, services.conversations]);
 
   useEffect(() => {
     services.conversations.load(bulkGetOptions);
+    const subscription = services.dataSource
+      .getDataSourceId$()
+      .pipe(skip(1))
```

Comment:
Could you leave a comment to describe why it should skip the first value?

Reply:
Sure, since the the conversations load will be called after mount. We skip the first value here to avoid duplicate load API request.

Reply:
Can we say that we are skipping the case that dataSource$ emits the same value? In that case we'd use `distinctUntilChanged` instead of skip the first value right?

Reply:
Not the same value. Since `dataSourceId$` and `defaultDataSourceId$` are `BehaviorSubject`.  The observer will be executed once first subscribed. The observer will reload conversations history. Since we already call `conversations.load` after mount, skip first value here to avoid duplicate API call.